qiskit_machine_learning/neural_networks/sampler_qnn.py

```diff
@@ -0,0 +1,343 @@
+
+    def __init__(
+        self,
+        *,
+        sampler: BaseSampler,
+        circuit: QuantumCircuit,
+        input_params: Sequence[Parameter] | None = None,
+        weight_params: Sequence[Parameter] | None = None,
+        sparse: bool = False,
+        interpret: Callable[[int], int | tuple[int, ...]] | None = None,
+        output_shape: int | tuple[int, ...] | None = None,
+        gradient: BaseSamplerGradient | None = None,
+        input_gradients: bool = False,
+    ):
+        """
+        Args:
+            sampler: The sampler primitive used to compute neural network's results.
+            circuit: The parametrized quantum circuit that generates the samples of this network.
+            input_params: The parameters of the circuit corresponding to the input.
+            weight_params: The parameters of the circuit corresponding to the trainable weights.
+            sparse: Returns whether the output is sparse or not.
+            interpret: A callable that maps the measured integer to another unsigned integer or
+                tuple of unsigned integers. These are used as new indices for the (potentially
+                sparse) output array. If no interpret function is
+                passed, then an identity function will be used by this neural network.
+            output_shape: The output shape of the custom interpretation
+            gradient: An optional sampler gradient to be used for the backward pass.
+            input_gradients: Determines whether to compute gradients with respect to input data.
+                 Note that this parameter is ``False`` by default, and must be explicitly set to
+                 ``True`` for a proper gradient computation when using ``TorchConnector``.
+        Raises:
+            QiskitMachineLearningError: Invalid parameter values.
+        """
+        # set primitive
+        self.sampler = sampler
+
+        # set gradient
+        # TODO: provide default gradient?
+        self.gradient = gradient or ParamShiftSamplerGradient(self.sampler)
+
+        self._circuit = circuit.copy()
+        if len(self._circuit.clbits) == 0:
+            self._circuit.measure_all()
+
+        self._input_params = list(input_params or [])
+        self._weight_params = list(weight_params or [])
+
+        if sparse:
+            _optionals.HAS_SPARSE.require_now("DOK")
+
+        self.set_interpret(interpret, output_shape)
+
+        self._input_gradients = input_gradients
+
+        # TODO: look into custom transpilation
+
+        super().__init__(
+            len(self._input_params),
+            len(self._weight_params),
+            sparse,
+            self._output_shape,
+            self._input_gradients,
+        )
+
+    @property
+    def circuit(self) -> QuantumCircuit:
+        """Returns the underlying quantum circuit."""
+        return self._circuit
+
+    @property
+    def input_params(self) -> Sequence:
```

Comment:
Sequence here is lacking types - Sequence{Parameter] right? Same for weight_params below